client/keys/add_test.go

```diff
@@ -211,7 +211,76 @@ func Test_runAddCmdMultisigDupKeys(t *testing.T) {
 	mockIn.Reset("y\n")
 	require.Error(t, cmd.ExecuteContext(ctx))
 	mockIn.Reset("y\n")
-	require.EqualError(t, cmd.ExecuteContext(ctx), "duplicate multisig keys: keyname1")
+	require.EqualError(t, cmd.ExecuteContext(ctx), "duplicate multisig keys found: keyname1")
+}
+
+func Test_runAddCmdMultisigMultipleDupKeys(t *testing.T) {
+	cmd := AddKeyCommand()
+	cmd.Flags().AddFlagSet(Commands().PersistentFlags())
+
+	mockIn := testutil.ApplyMockIODiscardOutErr(cmd)
+	kbHome := t.TempDir()
+
+	cdc := moduletestutil.MakeTestEncodingConfig(codectestutil.CodecOptions{}).Codec
+	kb, err := keyring.New(sdk.KeyringServiceName(), keyring.BackendTest, kbHome, mockIn, cdc)
+	require.NoError(t, err)
+
+	clientCtx := client.Context{}.
+		WithKeyringDir(kbHome).
+		WithInput(mockIn).
+		WithCodec(cdc).
+		WithAddressCodec(addresscodec.NewBech32Codec("cosmos")).
+		WithValidatorAddressCodec(addresscodec.NewBech32Codec("cosmosvaloper")).
+		WithConsensusAddressCodec(addresscodec.NewBech32Codec("cosmosvalcons"))
+
+	ctx := context.WithValue(context.Background(), client.ClientContextKey, &clientCtx)
+
+	t.Cleanup(func() {
+		_ = kb.Delete("key1")
+		_ = kb.Delete("key2")
+		_ = kb.Delete("key3")
+		_ = kb.Delete("multisigname")
+	})
```

Comment:
is this needed? you used `t.TempDir` so anything in that dir would be deleted after the test finishes.

Reply:
Your point makes sense. However, this is more of a personal coding habit.
Do you think “Cleanup” is unnecessary in cosmos-sdk’s testing approach?

Reply:
I think it would be better to address that issue in the next PR rather than in this one. What do you think?

Reply:
I think we can clean this PR up now

Reply:
Updated